nucliadb/src/nucliadb/search/augmentor/augmentor.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+import asyncio
+import re
+
+from nucliadb.common import ids
+from nucliadb.models.internal.augment import (
+    Augment,
+    FieldIdPattern,
+    ParagraphIdPattern,
+    ResourceIdPattern,
+)
+from nucliadb.search.augmentor.models import (
+    Augmented,
+    AugmentedField,
+    AugmentedParagraph,
+    AugmentedResource,
+)
+from nucliadb.search.augmentor.utils import limited_concurrency
+from nucliadb.search.search import cache
+
+from .fields import augment_field
+from .paragraphs import augment_paragraph
+from .resources import augment_resource
+
+
+async def augment(
+    kbid: str,
+    augmentations: list[Augment],
+    # TODO: limits
```

Comment:
what kind of limits?
These should be enforced at the pyndantic level maybe?

Reply:
It's meant to be global limits, like max number of images to augment or things like that, but I haven't implemented it yet. I can remove the TODO meanwhile